### PR TITLE
Update the donate modal

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/2260

- bottom should no longer be cut off on iPhone
- should no longer show pop up on campaign pages
- now has button hover state like on pni site

### CODE REVIEW

**Comment by kristinashu** [comment]
There aren't any campaign pages in this PR so I can't test that, so guess we'll see on staging.

The button isn't cut off but it still seems very close to the bottom. Not a blocker but wonder if you can make the `Chip in to help us keep the web healthy, wonderful, and welcoming to all.` the same font styling as the `Let's protect the world's...`. This way the text will be small and take up less room. This would be for the narrow width only. Do you think that would help?

screen shot 2018-11-22 at 2 38 27 pm

**Comment by Pomax** [comment]
I'd recommend tweaking the placement CSS next week, looking at each of the widths that bootstrap supports and seeing which special affordances are necessary.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/campaign_page.html [hunk 1]
```diff
@@ -1,6 +1,12 @@
 {% extends "./mini_site_name_space.html" %}
 {% load wagtailcore_tags mini_site_tags %}
 
+Note: we don't want the donation modal to pop up on
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/campaign_page.html]: should this be in a comment block?

File: source/js/donate-modal/donate-modal.scss [hunk 2]
```diff
@@ -81,6 +85,12 @@
     color: white;
     padding: 15px 18px;
     border-radius: calc((20px + 18px * 2) / 2); // half of button's height
+
+    &:focus,
```
Comment (mmmavis) [inline source/js/donate-modal/donate-modal.scss]: we can do

```scss
@include hover-focus-active {
  background: #f00;
}
```

File: source/js/donate-modal/donate-modal.scss [hunk 3]
```diff
@@ -97,5 +107,13 @@
     font-weight: 400;
 
     @include scaleText(22px, 28px, 22px, 28px);
+
+    @media (max-width: $bp-sm) {
+      font-family: "Nunito Sans", Helvetica, Arial, sans-serif;
+      font-size: 16px;
+      line-height: 1.5;
```
Comment (mmmavis) [inline source/js/donate-modal/donate-modal.scss]: we don't need
```css
font-size: 16px;
line-height: 1.5;
```
having the following line should be sufficient
```css
 @include scaleText(16px, 24px);
```

File: network-api/networkapi/wagtailpages/templates/wagtailpages/campaign_page.html (hunk 1 above)
Comment (Pomax) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/campaign_page.html]: As an `extends`, nothing outside of `{%block%}` codes is active code, so it doesn't have to be marked as comment. That said, trivial to add it in so why not.

File: source/js/donate-modal/donate-modal.scss (hunk 3 above)
Comment (Pomax) [reply source/js/donate-modal/donate-modal.scss]: oh nice!

**Pull request merged**